backend/src/main/java/org/eclipse/tractusx/puris/backend/common/edc/domain/model/ContractMapping.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2024 Volkswagen AG
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.eclipse.tractusx.puris.backend.common.edc.domain.model;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.Inheritance;
+import jakarta.persistence.InheritanceType;
+import lombok.*;
+
+@Entity
+@Getter
+@Setter
+@ToString
+@NoArgsConstructor
+@AllArgsConstructor
+@EqualsAndHashCode
+@Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
+public abstract class ContractMapping {
+
+    @Id
+    protected String partnerBpnl;
+    protected String contractId;
+    protected String assetId;
```

Comment:
With the current way things are implemented, we are expecting that the "one asset per submodel" is the most common way. And we are trying to leverage on that by trying to re-use previous contracts where ever possible, 

If that expectation is not fulfilled, then we are invalidating the stored contract information (i.E. contractId, assetId). 
If the same request is now done again, then there would be no stored contract data in the contract mapping service available. 

That means, in that repeated attempt, we would do a lookup specifically for the AAS that is related to the desired Material or Product.  

Which should contain exactly that one EDC asset ID that is propagated in the AAS. This is then stored. 

That means, if we knew in advance that a certrain submodel usually does not follow the "one asset per submodel" - approach, then it might be wiser to not integrate that submodel into the EdcContractMapping. 

However, for the possible case that in a submodel it happens occassionally (but not too often), that the assetId, that worked for Material A does not for Material B, I have included a retry-mechanism in the respective methods of the EdcAdapterServive. 

That means, with this addition we are effectively supporting all three cases mentioned above.

Reply:
Thanks for clarification and update! Resolved